paper-server/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java

```diff
@@ -375,7 +375,7 @@ public <T extends Entity> T addEntity(T entity) {
         Preconditions.checkArgument(!entity.isInWorld(), "Entity has already been added to a world");
         net.minecraft.world.entity.Entity nmsEntity = ((CraftEntity) entity).getHandle();
         if (nmsEntity.level() != this.getHandle().getLevel()) {
-            nmsEntity = nmsEntity.teleport(new TeleportTransition(this.getHandle().getLevel(), nmsEntity, TeleportTransition.DO_NOTHING));
+            throw new IllegalArgumentException(entity + " wasn't created with this world, you must create the entity with the world you want to add it to.");
         }
```

Comment:
i feel a Precondition can be good here if now this throw an exception.
```suggestion
        Preconditions.checkArgument(nmsEntity.level() == this.getHandle().getLevel(), "%s wasn't created with this world, you must create the entity with the world you want to add it to.", entity);
```

Reply:
Better to not evaluate that string earlier.